Sources/NIOSSH/SSHPacketParser.swift

```diff
@@ -114,7 +119,13 @@ struct SSHPacketParser {
     private mutating func readVersion() throws -> String? {
         // Looking for a string ending with \r\n
         let slice = self.buffer.readableBytesView
-        if let cr = slice.firstIndex(of: 13), cr.advanced(by: 1) < slice.endIndex, slice[cr.advanced(by: 1)] == 10 {
+        
+        // Prevent the consumed bytes for a version from exceeding the maximum packet size
+        // In practice, if SSH version packets come anywhere near this it's already likely an attack
+        // More data cannot be blindly regarded as malicious though, since these might be multiple packets
+        let maximumVersionSize = min(slice.endIndex, self.maximumPacketSize)
```

Comment:
`ByteBufferView` is not necessarily zero-indexed, so `self.maximumPacketSize` has to be translated into index-space first. I think this should be `let maxIndex = slice.index(slice.startIndex, offsetBy: min(slice.count, self.maximumPacketSize))`.